contracts/DegenLockToken.sol

```diff
@@ -0,0 +1,185 @@
+pragma solidity ^0.8.0;
+
+import {IERC20, SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+
+/**
+ * @notice Lock Degen for a set amount of time.
+ * @custom:security-contact jacek@degen.tips
+ */
+contract DegenLockToken is ERC20, Ownable, ReentrancyGuard {
+    using SafeERC20 for IERC20;
+
+    mapping(address account => uint256) public depositTimestamps;
+
+    /**
+     * @dev Name of the token representing the claim on the locked token
+     */
+    string public constant TOKEN_NAME = "Locked Degen";
+
+    /**
+     * @dev Symbol of the token representing the claim on the locked token
+     */
+    string public constant TOKEN_SYMBOL = "LDEGEN";
+
+    /**
+     * @dev The ERC20 token to be locked
+     */
+    IERC20 public constant TOKEN =
+        IERC20(0xfee293840D23B0B2De8C55e1Cf7A9F01C157767c);
+
+    /**
+     * @dev Lock duration in seconds, period starts after the deposit timestamp
+     */
+    uint256 public lockDuration = 90 days;
+
+    /**
+     * @dev Minimum amount of tokens that have to be deposited
+     */
+    uint256 public minDepositAmount = 1e22;
+
+    /**
+     * @dev Withdraw is not possible because the lock period is not over yet
+     */
+    error LockPeriodOngoing();
+
+    /**
+     * @dev ERC-20 function is not supported
+     */
+    error NotSupported();
+
+    /**
+     *  @dev The lock duration is longer than a year
+     */
+    error LockDurationIsTooLong();
+
+    /**
+     *  @dev Amount of tokens to be locked is zero
+     */
+    error ZeroAmount();
+
+    /**
+    /**
+     * @dev Minimum deposit not met
+     */
+    error MinimumDepositNotMet();
+
+    /**
+     *  @dev The lock duration has been updated
+     */
+    event LockDurationUpdated(uint256 duration);
+
+    /**
+     *  @dev The minimum deposit amount has been updated
+     */
+    event MinDepositAmountUpdated(uint256 minDepositAmount);
+
+    /**
+     *  @dev The deposit timestamp has been set
+     */
+    event DepositTimestampUpdated(
+        address indexed account,
+        uint256 depositTimestamp
+    );
+
+    /**
+     * @dev Construct a new Degen token
+     */
+    constructor() ERC20(TOKEN_NAME, TOKEN_SYMBOL) Ownable(msg.sender) {}
+
+    /**
+     * @dev Deposit tokens to be locked until the end of the locking period
+     * @param amount The amount of tokens to deposit
+     */
+    function deposit(uint256 amount) external nonReentrant {
+        if (amount < minDepositAmount) {
+            revert MinimumDepositNotMet();
+        }
+
+        TOKEN.safeTransferFrom(msg.sender, address(this), amount);
+        _mint(msg.sender, amount);
+
+        depositTimestamps[msg.sender] = block.timestamp;
+        emit DepositTimestampUpdated(msg.sender, block.timestamp);
```

Comment:
One minor thing:
It could be useful to include `amount` in the event parameter. Even though `_mint` will emit an event with the amount parameter already, tracking a single event that has all the parameters is more convenient than tracking two events.

Reply:
Great suggestion! I've created `Deposit` and `Withdraw` events for each function, logging both the sender address and the amount for each transaction. This should make tracking everything much more convenient.